modules/dataquery/jsx/definefields.tsx

```diff
@@ -0,0 +1,617 @@
+         </dl>
+         {visits}
+    </div>);
+}
+
+/**
+ * Render the define fields tab
+ *
+ * @param {object} props - React props
+ * @param {APIQueryField[]} props.selected - The currently selected fields
+ * @param {function} props.setSelected - Function to set the currently selected fields
+ * @param {function} props.onCategoryChange - Callback when the selected category changes
+ * @param {function} props.onFieldToggle - Callback when an item is clicked in the field list
+ * @param {function} props.onChangeVisitList - Callback when the visits are changed for a field
+ * @param {function} props.onChangeDefaultVisits - Callback when the default visits are changed
+ * @param {function} props.onAddAll - Callback when the "Add All" button is clicked
+ * @param {function} props.onRemoveAll - Callback when the "Remove All" button is clicked
+ * @param {function} props.onClearAll - Callback when the "Clear All" button is clicked
+ * @param {string} props.module - The module of the currently selected category
+ * @param {string} props.category - The currently selected category
+ * @param {FullDictionary} props.fulldictionary - The dictionary of all elements
+ * @param {function} props.removeField - callback to remove a single field
+ * @param {string[]} props.defaultVisits - The default visits
+ * @param {DictionaryCategory} props.displayedFields - The category currently selected to display
+ * @param {CategoriesAPIReturn} props.allCategories - all categories that exist
+ * @param {string[]} props.allVisits - All (non-phantom) visits for this LORIS instance
+ * @returns {React.ReactElement} - The Define Fields page
+ */
+function DefineFields(props: {
+    selected: APIQueryField[],
+    onCategoryChange: (module: string, category: string) => void,
+    setSelected: (newselected: APIQueryField[]) => void,
+    fulldictionary: FullDictionary,
+    displayedFields: DictionaryCategory,
+    removeField: (module: string, item: string, field: string) => void,
+    onClearAll: () => void,
+    defaultVisits: string[],
+    allCategories: CategoriesAPIReturn,
+    allVisits: string[],
+    onChangeDefaultVisits: (newvisits: readonly VisitOption[]) => void,
+    module: string,
+    category: string,
+    onRemoveAll: (removeelements: APIQueryField[]) => void,
+    onAddAll: (elements: APIQueryField[]) => void,
+    onChangeVisitList: (
+        module: string,
+        category: string,
+        field: string,
+        visits: string[]
+    ) => void,
+    onFieldToggle: (
+        module: string,
+        category: string,
+        field: string,
+        visits: string[]
+    ) => void,
+}) {
+  const [activeFilter, setActiveFilter] = useState('');
+  const [syncVisits, setSyncVisits] = useState<boolean>(false);
+  const [zoomTo, setZoomTo] = useState<string|null>(null);
+  useEffect(() => {
+      if (!syncVisits) {
+          return;
+      }
+      // FIXME: Go through each selected field, get the dictionary,
```

Comment:
Is this what's already being done here? Should the comment be removed / modified?